paddle/fluid/framework/details/multi_devices_graph_pass.cc

```diff
@@ -205,8 +205,9 @@ std::unique_ptr<ir::Graph> MultiDevSSAGraphBuilderBase::ApplyImpl(
             auto &p_name = backward_vars[i];
             auto &g_name = backward_vars[i + 1];
             VLOG(10) << "Bcast " << g_name << " for parameter " << p_name;
-
-            InsertCollectiveOp(&result, p_name, g_name);
+            if (NeedCollectiveOpHandle(g_name, sorted_ops)) {
```

Comment:
Merge `NeedCollectiveOpHandle ` and `NeedCollectiveOps ` ?

Reply:
Done.